admin/controller/altapay/templates/catalog/controller/altapay.twig.php

```diff
@@ -733,4 +744,22 @@ private function decodeHtmlEntitiesArrayValues($arr)
 
         return $arr;
     }
+
+    /**
+     * @param $order_id
+     * @param $post_data
+     *
+     * @return void
+     */
+    private function saveReconciliationIdentifier($order_id, $post_data)
+    {
+        $callback = new Callback($post_data);
+        $response = $callback->call();
+        if ($response && is_array($response->Transactions) && !empty($response->Transactions[0]->ReconciliationIdentifiers)) {
```

Comment:
Are you sure is always the first transaction?

What if this is a second capture, or a refund?

Reply:
Yes, because this function is only being used in the callbacks ok, notification & open. In these cases its always the first transaction if captured.